internal/controller/dnsrecord_controller.go

```diff
@@ -305,10 +305,22 @@ func (r *DNSRecordReconciler) publishRecord(ctx context.Context, dnsRecord *v1al
 			Namespace: dnsRecord.Namespace,
 		},
 	}
+
 	err := r.Get(ctx, client.ObjectKeyFromObject(managedZone), managedZone, &client.GetOptions{})
 	if err != nil {
 		return false, err
 	}
+
+	patchFrom := client.MergeFrom(dnsRecord.DeepCopy())
+	err = controllerutil.SetOwnerReference(managedZone, dnsRecord, r.Scheme)
```

Comment:
:+1: , you changed it before i got to comment. Is there no helper available  for checking owns? Also, this doesn't add the block on delete flag, do we care?

Reply:
yeah, thanks for the hint, haha! I found this: https://github.com/redhat-cop/operator-utils/blob/26a9d4efe6c40ed22f302342196ea6afc8b63170/pkg/util/owner.go#L24-L35 shall I add it as an import?